### PR TITLE
Header only build + cgal4h and BH deps

### PR DESCRIPTION
**Pull request by dickoa**
This is the first test to make allow your package to be built using CGAL header i.e no need to link the CGAL libraries. It works on Windows but for some reason (I see Rcpp on imports), Rcpp is not imported when the package is loaded. Other than this minor issues of Rcpp that need to imported it works well on Windows and can be built without any external libraries.
Let me know how it works
Ahmadou

### CODE REVIEW

**Comment by m-jahn** [comment]
Thanks so much, looks really promising! Unfortunately Travis reports a build error, but only due to timeout (>10 min) when running examples or vignette. The reason could be that it takes a bit longer than before, but I can test that. Because of the failed build I wouldn't merge the commit into master but into another branch to test it first. I just forwarded branch `windows` to master. Could you also merge your commit into that branch and make the PR? I will test building and performance and then merge master.

**Comment by m-jahn** [comment]
update: not building correctly in Travis check could be due to jump from R version 3.6 to 4.0, not the actual commit. Travis is running latest R version 4.0, while I checked package locally with 3.6. Need to test that.

**Comment by dickoa** [comment]
I will try to merge with the new branch and make a PR. In my local machine, the vignette took really long and I had to stop it, there's something not working and I didn't have the time check and dig into this.  Coming back to you with a PR later today.
Ahmadou

**Comment by m-jahn** [comment]
thanks for your patience. I'm investigating this right now, checking different R versions and will also check vignette building + examples. It should not take longer than 10 sec to  compute the example treemap, however Travis times out after 10 min trying to run examples..

**Comment by m-jahn** [comment]
I found the problem: some dependencies (Rcpp?) have changed over the past month, so after updating all my packages I could reproduce the error. On executing an example voronoi treemap generation, it would raise this error `function 'enterRNGScope' not provided by package 'Rcpp'` and be stuck forever. Travis then timed out after 10 min. A little googling revelaed that Rcpp wants to be explicitly imported, so I added `importFrom Rcpp evalCpp` as roxygen tag to `voronoiTreempa()` (gets then written to namespace when building package).
So now @dickoa please feel free to merge your changes with any but the master branch (devel or vignette). Thanks again!

**Comment by m-jahn** [comment]
Hey @dickoa, I just tested your changes locally, and it works. Really nice, it also erases the only remaining build warning about an `abort` statement in C++ code that was coming from CGAL library. One thing that did _not work_ for me though is the automatic installation of the `cgal4h` library just from the `Remotes:` line in `DESCRIPTION` file. Did that work for you if you _don't_ have the cgal4h package installed locally?
In other words, I had to install `cgal4h` manually, before building SysbioTreemaps in Rstudio.
I will merge your PR anyway because it is an improvement to not depend on CGAL library and only depend on R packages. Do you plan to submit `cgal4h` or `cgalh` to CRAN? Then I could submit my package too.
